### PR TITLE
add first jest unit tests and fix another bug in loadInitialProps

### PR DESCRIPTION
**Pull request by dagda1**
I have added tests around `loadInitialProps` after introducing a bug in a previous PR.

While writing these tests I discovered another bug in `loadInitialProps` where I was checking for the existence of `load` rather than `getInitialProps` before trying to resolve the call to `getInitialProps`.

I've fixed this and I've also renamed the types from `AsyncXXX` to `AfterXXX` because they might not be async and `AfterComponent` etc. seems more appropriate.

Apologies for the bugs but I don't think after has been published with these breaks and hopefully now the tests are in place, this should not happen again.

Is travis still a thing on github and should we get the CI going?

Be good if @stepanh could review also.

### CODE REVIEW

**Comment by dagda1** [comment]
hi, i can have a go at adding more tests if you think these are ok?

File: src/test/components/NonDynamicExport.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+
+class NonDymamiImport extends React.Component {
```
Comment (stepanh) [inline src/test/components/NonDynamicExport.tsx]: A little typo

File: src/types.ts [hunk 2]
```diff
@@ -24,35 +24,35 @@ export interface Ctx<P> extends CtxBase {
...
 }
 
-export interface AsyncRouteComponentState {
-  Component: AsyncRouteableComponent | null;
+export interface AfterRouteComponentState {
+  Component: AfterRouteableComponent | null;
 }
 
-export interface AsyncComponent {
+export interface AfterComponent {
```
Comment (stepanh) [inline src/types.ts]: Why did you rename this and other types?

Comment (dagda1) [reply src/types.ts]: @stepanh @jaredpalmer I could revert this, I just think `AfterXXXX` is a better name for some of the union types because they might not be async.

For example:

```
export type AfterRouteableComponent<Props = any> =
  | AfterRouteComponentType<RouteComponentProps<Props>>
  | React.ComponentType<RouteComponentProps<Props>>
  | React.ComponentType<Props>;
```

It might not have any async behaviour which makes the prefix confusing IMO.

@jaredpalmer is it a breaking change if it has not been published yet?

Comment (stepanh) [reply src/types.ts]: Makes sense, thanks for the clarification.

**Comment by dagda1** [comment]
@jaredpalmer is there anything more I need to address on this?

**Comment by jaredpalmer** [comment]
I don’t think we should rename components in the PR. Let’s add tests first. We can discuss naming after

**Comment by dagda1** [comment]
@jaredpalmer I've reverted the name changes

**Comment by jaredpalmer** [comment]
Will review

File: src/types.ts [hunk 3]
```diff
@@ -28,18 +28,18 @@ export interface AsyncRouteComponentState {
   Component: AsyncRouteableComponent | null;
 }
 
-export interface AsyncComponent {
```
[the remark this answers is not in the document]
Comment (dagda1) [reply src/types.ts]: @jaredpalmer renamed
